--- a/libmount/src/utils.c
+++ b/libmount/src/utils.c
@@ -276,14 +276,22 @@
 	return len;
 }
 
+#ifdef HAVE_STRUCT_STATX_STX_MNT_ID
 int get_mnt_id(const char *path, unsigned long *id)
 {
 	struct statx sx;
 	if (statx(AT_FDCWD, path, AT_SYMLINK_NOFOLLOW, STATX_MNT_ID, &sx) != 0)
 		return -1;
 	*id = sx.stx_mnt_id;
 	return 0;
 }
+#else
+int get_mnt_id(const char *path, unsigned long *id) { return -1; }
+#endif
 
 static int get_vfs_attr(int fd, const char *path, int *flags)
 {
